### ISSUE TITLE
Deletion Capabilities

### ISSUE THREAD

**Issue by Awk34** [Awk34/dotsynth#15]
Delete nodes, delete connections...

**Comment by Awk34** [comment]
The del(obj) method is partially implemented; need to create a trash can of sorts. Idea 1: drag dots onto trash can to delete them. Idea 2: click eraser button (eraser sticks to cursor), hover over a dot/connection (the dot/connection highlights), click/tap to delete it. What are your thoughts @uber5001 ?

**Comment by uber5001** [comment]
...we could do both. I'm not sure if that would be too confusing though.

Idea 1 makes sense for mobile devices. The user drags a dot around, and the trash can appears (or becomes highlighted in some way). I think most users will instantly figure out that they can drag the dot to the trash can. The only issue I have with this is that it might feel awkward dragging things to a trash can with a mouse. _But even desktop users will understand how it works_ (I think).

Idea 2 would only make sense for devices with a mouse. You need a mouse to hover and to change the cursor. The functionality is still good though.

So... I like idea 1 more, since it seems more intuitive. Idea 2 introduces a "mode" (deletion mode) which, without proper visual cues, can be confusing.

Ideally, I'd like both. I think idea 1 is easier to design.

**Comment by uber5001** [comment]
Oh, and for deleting connections, I'm thinking of 3 ways to do it.
1. In the dot's menu - have a list of all incoming about outgoing connections, whether it be to parameters or the node itself. Let users remove these connections from here (with a swipe gesture, perhaps).
2. Cut gesture - Hold and drag on the workspace to start a cut gesture. The cut gesture severs anyy connections it comes in contact with. Or...
3. Select gesture - Hold and drag on the workspace to begin a selection gesture. Touch any dots or connections with this gesture to "select" them. Any selected objects can be moved as a group, or removed, if the user taps the trash can with items selected. (There would need to be a visual cue on the trash can to indicate that the user can delete them.)

**Comment by Awk34** [comment]
1. I think just an 'X' button would be better for touch _and_ mouse controls
2. & 3. would interfere with our current new node menu, but 2. is good

**Comment by uber5001** [comment]
we're scrapping the current new node menu... right?

Plus, `addListeners` allows for both the new dot menu and one of those gestures. It can distinguish HoldDrag vs Hold.

**Comment by Awk34** [comment]
Definitely, just saying.

How? If I hold down without lifting my finger, the new node menu pops up. If I then drag, the menu stays there.

**Comment by uber5001** [comment]
Right. That's how it works... at the moment. It uses `onHoldStart`, which means as soon as enough time has passed, that menu appears. If you use `onHoldEnd` and `onHoldDrag*`, you can distinguish between the two.

**Comment by Awk34** [comment]
Okay just making sure you got that.
